custom/enikshay/case_utils.py

```diff
@@ -170,6 +172,39 @@ def get_open_episode_case_from_person(domain, person_case_id):
     )
 
 
+def get_open_referral_case_from_person(domain, person_case_id):
+    episode = get_open_episode_case_from_person(domain, person_case_id)
+    case_accessor = CaseAccessors(domain)
+    reverse_indexed_cases = case_accessor.get_reverse_indexed_cases([episode.case_id])
+    open_referral_cases = [
+        case for case in reverse_indexed_cases
+        if not case.closed and case.type == CASE_TYPE_REFERRAL
+    ]
+    if not open_referral_cases:
+        return None
+    if len(open_referral_cases) == 1:
+        return open_referral_cases[0]
+    else:
+        raise ENikshayException(
+            "Expected none or one open referral case for person with id: {}".format(person_case_id)
+        )
+
+
+def get_latest_trail_case_from_person(domain, person_case_id):
+    episode = get_open_episode_case_from_person(domain, person_case_id)
+    case_accessor = CaseAccessors(domain)
+    reverse_indexed_cases = case_accessor.get_reverse_indexed_cases([episode.case_id])
+    trail_cases = [
+        case for case in reverse_indexed_cases
+        if case.type == CASE_TYPE_TRAIL
+    ]
+    trail_cases.sort(key=lambda c: c.opened_on)
```

Comment:
What do you think of this for the a new case accessors method?
```python
    @staticmethod
    def get_newest_reverse_indexed_case(domain, case_ids, case_types):
        raise NotImplementedError
```
I'll probably also add `case_types` as an arg to `get_reverse_indexed_cases()`

Alternatively we could go very generic and add a `limit` and `order_by` param to `get_reverse_indexed_cases() `. Thoughts?